javascript/ql/src/semmle/javascript/DefensiveProgramming.qll

```diff
@@ -0,0 +1,412 @@
+/**
+ * Provides classes for working with defensive programming patterns.
+ */
+
+import javascript
+private import semmle.javascript.dataflow.InferredTypes
+
+/**
+ * A test in a defensive programming pattern.
+ */
+abstract class DefensiveExpression extends DataFlow::ValueNode {
```

Comment:
I'd prefer having the word "test" or "check" in the class name (it already appears in the qldoc comment).